### PR TITLE
[stable22]: Add composer patch

### PR DESCRIPTION
**Pull request by CarlSchwan**
This might be helpful later and unify the handling with master

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

**Comment by nickvergessen** [comment]
Conflicts with https://github.com/nextcloud/3rdparty/blob/stable22/patches.txt ? or well should we add them all to make sure they are applied correctly?

**Comment by CarlSchwan** [comment]
> Conflicts with https://github.com/nextcloud/3rdparty/blob/stable22/patches.txt ? or well should we add them all to make sure they are applied correctly?

Unfortunately, It seems this list of patches isn't getting applied for a while now :( And I'm not completely sure if we should try to apply them again or just remove that list and consider it as outdated.

**Comment by CarlSchwan** [comment]
I took a look at the patches listed in the file. These are the one who didn't get merged:

* https://github.com/doctrine/dbal/pull/3104
* https://github.com/fruux/sabre-dav/issues/982 dead link unfortunately
* https://github.com/fruux/sabre-xml/pull/132 but looking at the pull request an alternative to that one got merged in 2018
* https://github.com/McNetic/PHPZipStreamer/pull/39 but we now depend on the fork mentioned in the pull request

**Comment by nickvergessen** [comment]
https://github.com/fruux/sabre-dav/issues/982 is https://github.com/sabre-io/dav/pull/982

**Comment by nickvergessen** [comment]
> Fix like escape for sqlite doctrine/dbal#3104

When this is still applied in our code base we should add the patch as well, so it is reapplied on every update.
Otherwise good to go

File: cweagans/composer-patches/phpunit.xml.dist [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!--?xml version="1.0" encoding="UTF-8"?-->
```
Comment (nickvergessen) [inline cweagans/composer-patches/phpunit.xml.dist:1]: this and some other files should be added in gitignore instead

**Review by blizzz — APPROVED** [review]

**Pull request merged**